From fb0d3f589883053da8d46595e56dcfe42f571bf2 Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Mon, 16 Jul 2018 11:14:09 +0800
Subject: [PATCH] arm64: dts: rockchip: rename rk3326 evb v10 board for linux

This patch tried to rename arch/arm64/boot/dts/rockchip/
{rk3326-evb-linux-lp3-v10.dts => rk3326-evb-lp3-v10-linux.dts},
and correct the right things as below.

* define the rootfs type as the squashfs.
* fixes some typo
* copyright to 2018

Change-Id: I5fed2e472c979d5209222ad51281b01148379efb
Signed-off-by: Caesar Wang <wxt@rock-chips.com>
---
 .../devicetree/bindings/arm/rockchip.txt      |  2 +
 arch/arm64/boot/dts/rockchip/Makefile         |  2 +-
 ...3-v10.dts => rk3326-evb-lp3-v10-linux.dts} | 89 ++++++++++---------
 3 files changed, 51 insertions(+), 42 deletions(-)
 rename arch/arm64/boot/dts/rockchip/{rk3326-evb-linux-lp3-v10.dts => rk3326-evb-lp3-v10-linux.dts} (97%)

diff --git a/Documentation/devicetree/bindings/arm/rockchip.txt b/Documentation/devicetree/bindings/arm/rockchip.txt
index 869dbba0361e..fecd401ef7dd 100644
--- a/Documentation/devicetree/bindings/arm/rockchip.txt
+++ b/Documentation/devicetree/bindings/arm/rockchip.txt
@@ -122,6 +122,8 @@ Rockchip platforms device tree bindings
 - Rockchip RK3326 evb board:
     Required root node properties:
       - compatible = "rockchip,rk3326-evb-lp3-v10", "rockchip,rk3326";
+    Or
+      - compatible = "rockchip,rk3326-evb-lp3-v10-linux", "rockchip,rk3326";
 
 - Rockchip PX30 evb ddr3 board:
     Required root node properties:
diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index 6b1075591ae6..25661cd81e04 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -14,7 +14,7 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3308-voice-module-board-v10.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3326-evb-ai-va-v10.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3326-evb-ai-va-v11.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3326-evb-lp3-v10.dtb
-dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3326-evb-linux-lp3-v10.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3326-evb-lp3-v10-linux.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3326-863-lp3-v10.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3328-box-liantong.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3328-evb.dtb
diff --git a/arch/arm64/boot/dts/rockchip/rk3326-evb-linux-lp3-v10.dts b/arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10-linux.dts
similarity index 97%
rename from arch/arm64/boot/dts/rockchip/rk3326-evb-linux-lp3-v10.dts
rename to arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10-linux.dts
index 04d972316b38..9c296b0e417f 100644
--- a/arch/arm64/boot/dts/rockchip/rk3326-evb-linux-lp3-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3326-evb-lp3-v10-linux.dts
@@ -1,7 +1,6 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
 /*
- * Copyright (c) 2017 Fuzhou Rockchip Electronics Co., Ltd
- *
- * SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+ * Copyright (c) 2018 Fuzhou Rockchip Electronics Co., Ltd
  */
 
 /dts-v1/;
@@ -15,8 +14,12 @@
 #include "rk3326-linux.dtsi"
 
 / {
-	model = "Rockchip rk3326 evb linux board";
-	compatible = "rockchip,rk3326-evb-linux-lp3-v10", "rockchip,rk3326";
+	model = "Rockchip rk3326 evb lpddr3 v10 board for linux";
+	compatible = "rockchip,rk3326-evb-lp3-v10-linux", "rockchip,rk3326";
+
+	chosen {
+                bootargs = "earlycon=uart8250,mmio32,0xff160000 swiotlb=1 console=ttyFIQ0 root=PARTUUID=614e0000-0000 rootfstype=squashfs rootwait";
+        };
 
 	adc-keys {
 		compatible = "adc-keys";
@@ -832,42 +835,6 @@
 	};
 };
 
-&pinctrl {
-	headphone {
-		hp_det: hp-det {
-			rockchip,pins = <2 RK_PC6 RK_FUNC_GPIO &pcfg_pull_down>;
-		};
-	};
-
-	pmic {
-		pmic_int: pmic_int {
-			rockchip,pins =
-				<0 RK_PA7 RK_FUNC_GPIO &pcfg_pull_up>;
-		};
-
-		soc_slppin_gpio: soc_slppin_gpio {
-			rockchip,pins =
-				<0 RK_PA4 RK_FUNC_GPIO &pcfg_output_low>;
-		};
-
-		soc_slppin_slp: soc_slppin_slp {
-			rockchip,pins =
-				<0 RK_PA4 RK_FUNC_1 &pcfg_pull_none>;
-		};
-
-		soc_slppin_rst: soc_slppin_rst {
-			rockchip,pins =
-				<0 RK_PA4 RK_FUNC_2 &pcfg_pull_none>;
-		};
-	};
-
-	sdio-pwrseq {
-		wifi_enable_h: wifi-enable-h {
-			rockchip,pins = <0 RK_PA2 RK_FUNC_GPIO &pcfg_pull_none>;
-		};
-	};
-};
-
 &pmu_io_domains {
 	status = "okay";
 
@@ -978,3 +945,43 @@
 &vpu_combo {
 	status = "okay";
 };
+
+&pinctrl {
+	headphone {
+		hp_det: hp-det {
+			rockchip,pins = <2 RK_PC6 RK_FUNC_GPIO &pcfg_pull_down>;
+		};
+	};
+
+	pmic {
+		pmic_int: pmic_int {
+			rockchip,pins =
+				<0 RK_PA7 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+
+		soc_slppin_gpio: soc_slppin_gpio {
+			rockchip,pins =
+				<0 RK_PA4 RK_FUNC_GPIO &pcfg_output_low>;
+		};
+
+		soc_slppin_slp: soc_slppin_slp {
+			rockchip,pins =
+				<0 RK_PA4 RK_FUNC_1 &pcfg_pull_none>;
+		};
+
+		soc_slppin_rst: soc_slppin_rst {
+			rockchip,pins =
+				<0 RK_PA4 RK_FUNC_2 &pcfg_pull_none>;
+		};
+	};
+
+	sdio-pwrseq {
+		wifi_enable_h: wifi-enable-h {
+			rockchip,pins = <0 RK_PA2 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+};
+
+/* DON'T PUT ANYTHING BELOW HERE.  PUT IT ABOVE PINCTRL */
+/* DON'T PUT ANYTHING BELOW HERE.  PUT IT ABOVE PINCTRL */
+/* DON'T PUT ANYTHING BELOW HERE.  PUT IT ABOVE PINCTRL */
-- 
2.35.3

